sql/planbuilder/builder.go

```diff
@@ -232,6 +232,7 @@ func (b *Builder) buildSubquery(inScope *scope, stmt ast.Statement, subQuery str
 	case *ast.Show:
 		return b.buildShow(inScope, n)
 	case *ast.DDL:
+		b.qFlags.Set(sql.QFlagDDL)
```

Comment:
This is a good start but incomplete. There are other DDL operations that aren't captured by this node, notably ALTER TABLE

Reply:
Should add more tests for other statement types

Reply:
Included `QFlagAlterTable` and added tests for `alter table` and `create index`